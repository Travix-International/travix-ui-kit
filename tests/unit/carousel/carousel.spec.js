import { mount } from 'enzyme';
import React from 'react';
import Carousel from '../../../components/carousel/carousel';

describe('Carousel', () => {
  const requestAnimationFrame = global.window.requestAnimationFrame;

  beforeEach(() => {
    global.window.requestAnimationFrame = (cb) => { cb(); };
  });

  afterEach(() => {
    global.window.requestAnimationFrame = requestAnimationFrame;
  });

  describe('#render()', () => {
    it('render with default props and mods', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} mods={['small']} />
      );

      const items = renderTree.find('.ui-carousel-item');

      expect(renderTree).toMatchSnapshot();
      expect(renderTree.children('.ui-carousel').hasClass('ui-carousel_small')).toEqual(true);
      expect(items.length).toEqual(images.length);
    });

    it('show correct pagination info', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} />
      );

      const pagination = renderTree.find('.ui-carousel-page');

      expect(renderTree).toMatchSnapshot();
      expect(pagination.text()).toEqual(`1 / ${images.length}`);

      const nextButton = renderTree.find('.ui-carousel-navigation > a').last();
      nextButton.simulate('click');
      expect(pagination.text()).toEqual(`2 / ${images.length}`);
    });

    it('hitting previous move to last item', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} />
      );

      expect(renderTree).toMatchSnapshot();

      const pagination = renderTree.find('.ui-carousel-page');
      const prevButton = renderTree.find('.ui-carousel-navigation > a').first();
      const nextButton = renderTree.find('.ui-carousel-navigation > a').last();
      nextButton.simulate('click');
      prevButton.simulate('click');
      prevButton.simulate('click');
      expect(pagination.text()).toEqual(`${images.length} / ${images.length}`);
    });

    it('hitting next move to first item', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} />
      );

      expect(renderTree).toMatchSnapshot();

      const pagination = renderTree.find('.ui-carousel-page');
      const prevButton = renderTree.find('.ui-carousel-navigation > a').last();
      prevButton.simulate('click');
      prevButton.simulate('click');
      prevButton.simulate('click');
      expect(pagination.text()).toEqual(`1 / ${images.length}`);
    });

    it('display markers instead of pagination', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} markers />
      );

      expect(renderTree).toMatchSnapshot();

      const pagination = renderTree.find('.ui-carousel-page');
      const markers = renderTree.find('.ui-carousel-markers');
      expect(pagination.exists()).toEqual(false);
      expect(markers.exists()).toEqual(true);
    });

    it('clicking markers to change pagination', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} markers />
      );

      expect(renderTree).toMatchSnapshot();

      const markers = renderTree.find('.ui-carousel-markers');
      const markItem = markers.find('button').last();

      markItem.simulate('click');
      expect(renderTree.instance().state.currentItem).toEqual(images.length - 1);
    });

    it('swiping will change the current item', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} markers />
      );
      expect(renderTree).toMatchSnapshot();

      const instance = renderTree.instance();

      instance.handleSwipeNext();
      expect(instance.state.currentItem).toEqual(1);

      instance.handleSwipePrev();
      expect(instance.state.currentItem).toEqual(0);
    });

    it('prevent swipe before first item', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} markers />
      );
      expect(renderTree).toMatchSnapshot();

      const instance = renderTree.instance();
      instance.handleSwipePrev();

      expect(instance.state.currentItem).toEqual(0);
    });

    it('prevent swipe after last item', () => {
      const images = [
        'http://lorempixel.com/600/400/city',
        'http://lorempixel.com/600/400/sports',
        'http://lorempixel.com/600/400/people',
      ];
      const renderTree = mount(
        <Carousel images={images} markers />
      );
      expect(renderTree).toMatchSnapshot();

      const instance = renderTree.instance();

      instance.handleSwipeNext();
      instance.handleSwipeNext();
      instance.handleSwipeNext();

      expect(instance.state.currentItem).toEqual(images.length - 1);
    });
  });
});
