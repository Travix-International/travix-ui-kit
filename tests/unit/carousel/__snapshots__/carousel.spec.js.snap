// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel #render() clicking markers to change pagination 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={true}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselMarkers
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
        onClick={[Function]}
      >
        <div
          className="ui-carousel-markers"
        >
          <span />
          <button
            data-index={1}
            onClick={[Function]}
          />
          <button
            data-index={2}
            onClick={[Function]}
          />
        </div>
      </CarouselMarkers>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() display markers instead of pagination 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={true}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselMarkers
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
        onClick={[Function]}
      >
        <div
          className="ui-carousel-markers"
        >
          <span />
          <button
            data-index={1}
            onClick={[Function]}
          />
          <button
            data-index={2}
            onClick={[Function]}
          />
        </div>
      </CarouselMarkers>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() hitting next move to first item 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={false}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselPage
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
      >
        <div
          className="ui-carousel-page"
        >
          1
           / 
          3
        </div>
      </CarouselPage>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() hitting previous move to last item 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={false}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselPage
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
      >
        <div
          className="ui-carousel-page"
        >
          1
           / 
          3
        </div>
      </CarouselPage>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() prevent swipe after last item 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={true}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselMarkers
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
        onClick={[Function]}
      >
        <div
          className="ui-carousel-markers"
        >
          <span />
          <button
            data-index={1}
            onClick={[Function]}
          />
          <button
            data-index={2}
            onClick={[Function]}
          />
        </div>
      </CarouselMarkers>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() prevent swipe before first item 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={true}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselMarkers
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
        onClick={[Function]}
      >
        <div
          className="ui-carousel-markers"
        >
          <span />
          <button
            data-index={1}
            onClick={[Function]}
          />
          <button
            data-index={2}
            onClick={[Function]}
          />
        </div>
      </CarouselMarkers>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() render with default props and mods 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={false}
  mods={
    Array [
      "small",
    ]
  }
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel ui-carousel_small"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselPage
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
      >
        <div
          className="ui-carousel-page"
        >
          1
           / 
          3
        </div>
      </CarouselPage>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() show correct pagination info 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={false}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselPage
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
      >
        <div
          className="ui-carousel-page"
        >
          1
           / 
          3
        </div>
      </CarouselPage>
    </div>
  </div>
</Carousel>
`;

exports[`Carousel #render() swiping will change the current item 1`] = `
<Carousel
  current={0}
  images={
    Array [
      "http://lorempixel.com/600/400/city",
      "http://lorempixel.com/600/400/sports",
      "http://lorempixel.com/600/400/people",
    ]
  }
  markers={true}
  nextButton={
    <span>
      ›
    </span>
  }
  prevButton={
    <span>
      ‹
    </span>
  }
>
  <div
    className="ui-carousel"
  >
    <Swipe
      onLeft={[Function]}
      onRight={[Function]}
      style={
        Object {
          "transform": "translateX(-0%)",
        }
      }
    >
      <Touch
        className="ui-swipe"
        onEnd={[Function]}
        onLeft={[Function]}
        onMove={[Function]}
        onRight={[Function]}
        onStart={[Function]}
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="ui-swipe"
          onLeft={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
          onRight={[Function]}
          style={
            Object {
              "transform": "translateX(-0%)",
            }
          }
        >
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/city"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/city')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={true}
            src="http://lorempixel.com/600/400/sports"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={
                  Object {
                    "backgroundImage": "url('http://lorempixel.com/600/400/sports')",
                  }
                }
              >
                 
              </div>
            </div>
          </CarouselItem>
          <CarouselItem
            load={false}
            src="http://lorempixel.com/600/400/people"
          >
            <div
              className="ui-carousel-item"
            >
              <div
                className="ui-carousel-item-image"
                style={Object {}}
              >
                 
              </div>
            </div>
          </CarouselItem>
        </div>
      </Touch>
    </Swipe>
    <div>
      <div
        className="ui-carousel-navigation"
      >
        <button
          onClick={[Function]}
        >
          <span>
            ‹
          </span>
        </button>
        <button
          onClick={[Function]}
        >
          <span>
            ›
          </span>
        </button>
      </div>
      <CarouselMarkers
        current={0}
        images={
          Array [
            "http://lorempixel.com/600/400/city",
            "http://lorempixel.com/600/400/sports",
            "http://lorempixel.com/600/400/people",
          ]
        }
        onClick={[Function]}
      >
        <div
          className="ui-carousel-markers"
        >
          <span />
          <button
            data-index={1}
            onClick={[Function]}
          />
          <button
            data-index={2}
            onClick={[Function]}
          />
        </div>
      </CarouselMarkers>
    </div>
  </div>
</Carousel>
`;
